packages/lib-utils/src/components/table/VirtualizedTableBody.tsx

```diff
@@ -0,0 +1,129 @@
+import type { AnyObject } from '@monorepo/common';
+import type { ICell, SortByDirection, ThProps } from '@patternfly/react-table';
+import { VirtualTableBody } from '@patternfly/react-virtualized-extension';
+import type { Scroll } from '@patternfly/react-virtualized-extension/dist/js/components/Virtualized/types';
+import * as React from 'react';
+import { CellMeasurerCache, CellMeasurer } from 'react-virtualized';
+
+export type RowProps<D> = {
+  /** Row data object. */
+  obj: D;
+};
+
+export type TableColumn<D> = ICell & {
+  /** Column ID. */
+  id: string;
+  /** Optional sort configuration. */
+  sort?: ((data: D[], sortDirection: SortByDirection) => D[]) | ThProps['sort'] | string;
+  /** Optional visibility. */
+  visibility?: string[];
+};
+
+export type TableRowProps = {
+  /** Row ID. */
+  id: string;
+  /** Row index. */
+  index: number;
+  /** Optional row style. */
+  style: object;
+  /** Row key. */
+  trKey: string;
+  /** Optional className. */
+  className?: string;
+};
+
+export const TableRow: React.FC<TableRowProps> = ({ id, style, trKey, className }) => {
+  return <tr id={id} style={style} key={trKey} className={className} role="row" />;
+};
+
+type VirtualizedTableBodyProps<D> = {
+  /** Table columns. */
+  columns: TableColumn<D>[];
+  /** Data to be rendered. */
+  data: D[];
+  /** Table body height. */
+  height: number;
+  /** isScrolling flag. */
+  isScrolling: boolean;
+  /** onChildScroll callback. */
+  onChildScroll: (params: Scroll) => void;
+  /** Row component. */
+  Row: React.ComponentType<RowProps<D>>;
+  /** Scroll top number. */
+  scrollTop: number;
+  /** Table body width. */
+  width: number;
+};
+
+const RowMemo = React.memo<RowProps<any> & { Row: React.ComponentType<RowProps<any>> }>(
+  ({ Row, obj }: RowProps<any> & { Row: React.ComponentType<RowProps<any>> }) => <Row obj={obj} />,
+);
+
+const VirtualizedTableBody = <D,>({
+  columns,
+  data,
+  height,
+  isScrolling,
+  onChildScroll,
+  Row,
+  scrollTop,
+  width,
+}: VirtualizedTableBodyProps<D>) => {
+  const cellMeasurementCache = new CellMeasurerCache({
+    fixedWidth: true,
+    minHeight: 44,
```

Comment:
Curious where we got 44 from?

Reply:
@florkbr yeah, that's a bit weird that there is such a hardcoded value.
It's used in the official example https://github.com/patternfly/patternfly-react/blob/main/packages/react-virtualized-extension/src/components/Virtualized/examples/VirtualizedTable.md
and both https://github.com/openshift/console/blob/33f02a8f454519e10fd498a3037e25c26f3d0c0a/frontend/public/components/factory/Table/VirtualizedTableBody.tsx#L44 and https://github.com/openshift/hac-dev/blob/eb1864d25c7a235838c0c1c338b8abf1b356a7fe/src/shared/components/table/VirtualBody.tsx#L47